eth/stagedsync/exec3.go

```diff
@@ -900,8 +902,21 @@ Loop:
 					}
 					checkReceipts := !cfg.vmConfig.StatelessExec && chainConfig.IsByzantium(txTask.BlockNum) && !cfg.vmConfig.NoReceipts && !isMining
 					if txTask.BlockNum > 0 && !skipPostEvaluation { //Disable check for genesis. Maybe need somehow improve it in future - to satisfy TestExecutionSpec
-						if err := core.BlockPostValidation(usedGas, blobGasUsed, checkReceipts, txTask.BlockReceipts, txTask.Header, isMining); err != nil {
-							return fmt.Errorf("%w, txnIdx=%d, %v", consensus.ErrInvalidBlock, txTask.TxIndex, err) //same as in stage_exec.go
+						// This is an optimization to avoid running the post validation for every transaction in the block serially
+						// We only run it once for the last transaction in the block, this yields a +5% improvement in block processing time
+						if blockNum == maxBlockNum {
```

Comment:
I don't understand this comment: `every transaction in the block`
- we are already in `txTask.Final` section. so, how it can be `every transaction`? 
- `if blockNum == maxBlockNum` says me that you probably mean `every block`?  
- `+5% improvement in block processing time` - even if you exec 1 block on chain tip?